### PR TITLE
fix: validate that NCIDs are well-formed GUIDs

### PR DESCRIPTION
**Pull request by rbtr**
**Reason for Change**:

Validate that the input requests are for valid NCIDs, otherwise don't attempt to process them.

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] relevant PR labels added

**Notes**:

### CODE REVIEW

**Review by thatmattlong — APPROVED** [review]

File: cns/restserver/api.go [hunk 1]
```diff
@@ -786,14 +786,19 @@ func (service *HTTPRestService) createOrUpdateNetworkContainer(w http.ResponseWr
...
+	if err := service.Listener.Decode(w, r, &req); err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+	if err := req.Validate(); err != nil {
+		w.WriteHeader(http.StatusBadRequest)
 		return
 	}
 
+	logger.Request(service.Name, req.String(), nil)
```
[the remark this answers is not in the document]
Comment (rbtr) [reply cns/restserver/api.go:798]: we are logging [_in the decode function_ if there is an error already](https://github.com/Azure/azure-container-networking/blob/36c96ad3d2b44ebe986f4b77ffd09c195596cd11/common/listener.go#L146C71-L146C71), so this was effectively a dupe log before.

File: cns/restserver/api_test.go [hunk 2]
```diff
@@ -105,15 +105,15 @@ var (
...
+		ncID:         "f47ac10b-58cc-0372-8567-0e02b2c3d477",
 		ncIP:         "10.0.0.5",
 		ncType:       cns.AzureContainerInstance,
 		ncVersion:    "0",
 		podName:      "testpod",
 		podNamespace: "testpodnamespace",
 	}
 	nc4 = createOrUpdateNetworkContainerParams{
-		ncID:         "2abc",
+		ncID:         "f47ac10b-58cc-0372-8567-0e02b2c3d476",
```
Comment (tamilmani1989) [inline cns/restserver/api_test.go:116]: can we have one testcase with swift prefix?

File: cns/restserver/api.go [hunk 3]
```diff
@@ -786,14 +786,19 @@ func (service *HTTPRestService) createOrUpdateNetworkContainer(w http.ResponseWr
...
 
 	var req cns.CreateNetworkContainerRequest
-	err := service.Listener.Decode(w, r, &req)
-	logger.Request(service.Name, req.String(), err)
-	if err != nil {
+	if err := service.Listener.Decode(w, r, &req); err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+	if err := req.Validate(); err != nil {
```
Comment (tamilmani1989) [inline cns/restserver/api.go:793]: can we log if validation failed?

File: cns/NetworkContainerContract_test.go [hunk 4]
```diff
@@ -105,3 +105,40 @@ func TestNewPodInfoFromIPConfigsRequest(t *testing.T) {
 		})
 	}
 }
+
+func TestCreateNetworkContainerRequest_Validate(t *testing.T) {
```
Comment (tamilmani1989) [inline cns/NetworkContainerContract_test.go:109]: should we have another UT for postnetworkcontainerrequest validate?

File: cns/restserver/util.go [hunk 5]
```diff
@@ -955,6 +950,10 @@ func (service *HTTPRestService) handlePostNetworkContainers(w http.ResponseWrite
 		logger.Response(service.Name, response, response.Response.ReturnCode, err)
 		return
 	}
+	if err := req.Validate(); err != nil { //nolint:govet // shadow okay
+		w.WriteHeader(http.StatusBadRequest)
```
Comment (tamilmani1989) [inline cns/restserver/util.go:954]: log if validation failed?

File: cns/restserver/api_test.go (hunk 2 above)
Comment (rbtr) [reply cns/restserver/api_test.go:116]: no, the swift prefix is already added during the relevant tests. this represents the NMA NCID which will never be prefixed

File: cns/NetworkContainerContract_test.go (hunk 4 above)
Comment (rbtr) [reply cns/NetworkContainerContract_test.go:109]: that really only tests that the for loop works as expected, but sure

File: cns/restserver/api.go (hunk 3 above)
Comment (rbtr) [reply cns/restserver/api.go:793]: added

**Review by tamilmani1989 — APPROVED** [review]
lgtm

**Review by msvik — APPROVED** [review]

**Pull request merged**